### PR TITLE
Profile and sign out functionality

### PR DESCRIPTION
**Pull request by beaverseven**
Added a profile page that lets users see their email and allows users to change their password. Added a sign out functionality as well on the top of the navbar, bringing users back to the login/signup page.
Changes:
-Added Profile.js. Fetches user data from firebase initially and displays their email address and allows users to change their password. This is because we want users to have flexibility in their accounts and be able to view their account info.
-Added AvatarUpload. Something temporary if we want to pursue uploading our own avatars to profiles for review purposes. 
-UpdatePassword.jsx. A file that allows users to change their password through the firebase changeUserPassword method. This was added because users can forget their password or might want to change it to make it more secure.
-Changes to AppNavbar. Added a logout button to the navbar once users are logged in. Additionally, when users are logged in, the navbar changes. This change was made in order to add functionality to the app.

A lot of changes for one PR, I understand. But I was so caught up in getting this actually functioning together.

### CODE REVIEW

File: my-app/src/Components/Profile.js
Comment (chizy7) [inline my-app/src/Components/Profile.js:1]: It seems that there's a logical error in the code. If the user is not found, the `usernotfound` state should be set to `true`. The current code sets it to `true` when the user is found. Update the condition in the if statement within the `useEffect` hook:
```
if (!userDoc.exists()) {
  setUsernotfound(true);
} else {
  setUserData(userDoc.data());
}
```

Update the condition to display the "User not found" message when `usernotfound` is `true`:
```
if (usernotfound) {
  return <div>User not found</div>;
}
```

File: my-app/src/Components/ProfilePage/AvatarUpload.jsx
Comment (chizy7) [inline my-app/src/Components/ProfilePage/AvatarUpload.jsx:1]: This looks good and seem to work well. 
Just a few suggestions for better improvement:

1. Maybe rename the `UploadImage` function to `AvatarUpload` to match the filename and maintain consistency
2. Wrap the file input and button inside a form element for better accessibility and to allow for form submission using the Enter key:
```
return (
  <form>
    <input type="file" onChange={handleChange} />
    <button type="submit" onClick={handleUpload}>Upload</button>
  </form>
);
```
3. Prevent the default form submission behavior by adding `event.preventDefault()` inside the `handleUpload` function:
```
const handleUpload = (event) => {
  event.preventDefault();
  // ...
};
```
4. Add error handling and user feedback to inform users about the success or failure of their upload and display the message to the user.

File: my-app/src/Components/ProfilePage/UpdatePassword.jsx
Comment (chizy7) [inline my-app/src/Components/ProfilePage/UpdatePassword.jsx:1]: The UpdatePassword.jsx file looks well-implemented, making use of react-hook-form along with yup for form validation. However, there are a few minor suggestions to improve the code:

1. Rename the `PasswordResetForm` function to `UpdatePassword` to match the filename and maintain consistency:
```
export default function UpdatePassword() {
  // ...
}
```
2. Add a `minLength` constraint to the password field for better security:
```
const schema = yup.object().shape({
  password: yup.string().required().min(8, "Password must be at least 8 characters"),
});
```
3. Display a user-friendly error message when an error occurs during the password update process:
```
if (res?.error) setError(res.error.message)
```

File: my-app/src/Components/ProfilePage/UserInfo.jsx
Comment (chizy7) [inline my-app/src/Components/ProfilePage/UserInfo.jsx:1]: The UserInfo.jsx file is simple and seems to work correctly. However, there is an empty string in the `className` attribute, which should be removed or replaced with a class name if needed, unless there is a reason why you have it there, you can clarify for me. 
Additionally, the commented-out lines can be removed if they are not necessary?

File: my-app/src/firebase.js
Comment (chizy7) [inline my-app/src/firebase.js:1]: The firebase.js file seems to be correct, and it's handling the sign out and update password functionality. However, you could improve the error handling by logging the error messages in the catch block to make it easier to debug any issues that may arise.

**Review by chizy7 — CHANGES_REQUESTED** [review]
Great job! You can make the minor changes for better improvements. Otherwise, everything looks good. Thanks!

**Review by caid4134 — APPROVED** [review]
Looks good overall.

**Review by caid4134 — APPROVED** [review]
Changes look good overall.

**Pull request merged**